fix(video): fix toggle video bugs, improve video unit tests

maljuburi opened the pull request:
**What this PR does** 📖
- Fix toggle video bugs.
- improve unit test for video actions and mutations, unit test whether `toggleCamera` indeed toggles the `disabled` state of the video.

**Which issue(s) this PR fixes** 🔨
AP-1620

molimauro commented on components/views/media/actions/Actions.vue:
```diff
@@ -54,10 +54,13 @@ export default Vue.extend({
       this.isLoading = true
       try {
         if (kind === 'audio') {
           this.$store.dispatch('audio/toggleMute', {}, { root: true })
```
I know its not related to this but can we remove the `, {}, { root: true }` also here? Thanks!

maljuburi commented:
> Just 2 minor ones :)

Thanks @molimauro for the review.. I have addressed your feedback and also improved the test suit.. I have mocked the `call.mute` and `call.unmute` to make sure we are correctly calling them, and resolved some conflicts related to recent merge.

stavares843 commented:
thanks @maljuburi 🎉 btw feel free to change labels as well when you address the review feedback 🔨

maljuburi commented:
> thanks @maljuburi 🎉 btw feel free to change labels as well when you address the review feedback 🔨

got it! Thanks @stavares843

tooshel commented on components/views/navigation/sidebar/controls/Controls.vue:
```diff
@@ -81,4 +81,5 @@ export default Vue.extend({
 </script>
 
-<style scoped lang="less" src="./Controls.less"></style>
+<style scoped lang="less" src="./Controls.less">
```
Linter?

maljuburi replied:
Yup

tooshel commented on store/webrtc/actions.ts:
```diff
@@ -459,10 +459,13 @@ const webRTCActions = {
+          kind === 'audio' ? rootState.audio.muted : rootState.video.disabled,
       })
-      if (rootState.audio.muted) {
+      if (kind === 'audio' && rootState.audio.muted) {
```
Why do we use backticks instead of regular ' or " . . . is that a vue thing?

WanderingHogan replied:
these are single quotes?

tooshel replied:
they are . . . but they looked like backticks to me for a minute. When I looked in the code they were right but github UI was messing with me.

tooshel reviewed (approved):
Just a few comments of me wondering stuff but it looks good!

maljuburi commented on components/views/navigation/sidebar/controls/Controls.vue:
```diff
@@ -63,9 +63,9 @@ export default Vue.extend({
       this.isLoading = true
       try {
         if (kind === WebRTCEnum.AUDIO) {
-          this.$store.dispatch('audio/toggleMute', undefined, { root: true })
```
I was wondering why we had it there too since the `toggleMute` is already aware of the state. good callout, I just went ahead and changed few things in the `audio/action` and committed [here](https://github.com/Satellite-im/Core-PWA/pull/3490/commits/7406c41fbf189e9e9b82cc6e1af9e90279be8a07) . hopefully that doesn't break something that I am not aware of. but I tested the audio and still work as expected.

molimauro commented:
Thanks @maljuburi great job!

phillsatellite commented:
Little bug I was able to replicate, it seems like if both users are sharing video feed, after a few moments it will cut off one of the video feeds but the button will still appear as though your camera is on
https://user-images.githubusercontent.com/93608357/172893515-940180ad-817a-44bf-ac68-67e0c0fa0727.mov

stavares843 commented:
<img width="888" alt="Screenshot 2022-06-09 at 17 11 33" src="https://user-images.githubusercontent.com/29093946/172894394-6a5c274d-3078-44f0-9f8c-9e3211ab7b49.png">

tests are failing on this PR, let us know if you need any help @maljuburi 🔨 our SDET can also take a look

maljuburi commented:
> Little bug I noticed, it seems like if both users are sharing video feed, after a few moments it will cut off one of the video feeds but the button will still appear as though your camera is on
> 
>  both.videos.mov

race condition issue... I noticed that yesterday too and mentioned it one of our teams call.. I am taking a look. Thanks @phillsatellite

maljuburi commented:
> <img alt="Screenshot 2022-06-09 at 17 11 33" width="888" src="https://user-images.githubusercontent.com/29093946/172894394-6a5c274d-3078-44f0-9f8c-9e3211ab7b49.png">
> 
> tests are failing on this PR, let us know if you need any help @maljuburi 🔨 our SDET can also take a look

audio unit test has updated since the last dev rebase I did locally, rebasing again to fix it. good catch, thanks

stavares843 commented on store/video/actions.test.ts:
```diff
@@ -8,15 +8,22 @@ import {
-//     return false
-//   }
-// })
+jest.mock('~/libraries/WebRTC/WebRTC', () => ({
```
neat 🎉

stavares843 replied:
neat 🎉

stavares843 commented:
thanks for updating the tests @maljuburi 🎉

stavares843 commented on components/views/media/actions/Actions.vue:
```diff
@@ -57,12 +57,13 @@ export default Vue.extend({
      * @example
      */
     toggleMute(kind: WebRTCEnum) {
+      // TODO: isLoading needs to be kind specific, currently all 3 kinds show loading icon if any of them is loading.
```
added task [here](https://satellite-im.atlassian.net/browse/AP-1774)
